lib-js/reporter.ts

```diff
@@ -0,0 +1,81 @@
+import {Writable} from 'stream';
+
+/**
+ * This reporter is used by linters to report findings and add annotations used
+ * by GitHub actions to display enhance the output, for example making paths to
+ * the problematic files link to the file in a GitHub repo.
+ *
+ * This reporter will print out a notice suggesting users to add the flag
+ * `--fix` on their linter to automatically fix the linter findings reported
+ * as being fixable.
+ */
+export class GithubActionsReporter {
```

Comment:
I think the original name, `Reporter`, is better—although this *can* print GitHub Actions annotations, it also runs as just a normal human-friendly reporter if the `GITHUB_ACTIONS` environment variable isn't set.